nilearn/_utils/niimage.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import gc
+import collections
+
+import nibabel
+
+
+def _get_shape(img):
+    # Use the fact that Nifti1Image has a shape attribute that is
+    # faster than loading the data from disk
+    if hasattr(img, 'shape'):
+        shape = img.shape
+    else:
+        shape = img.get_data().shape
+    return shape
+
+
+def _safe_get_data(img):
+    """ Get the data in the image without having a side effect on the
+        Nifti1Image object
+    """
+    if hasattr(img, '_data_cache') and img._data_cache is None:
+        # Copy locally the Nift1Image object to avoid the side effect of data
+        # loading
+        img = copy.deepcopy(img)
+    # typically the line below can double memory usage
+    # that's why we invoke a forced call to the garbage collector
+    gc.collect()
+    return img.get_data()
+
+
+def is_img(obj):
+    """ Check for get_data and get_affine method in an object
+
+    Parameters
+    ----------
+    obj: any object
+        Tested object
+
+    Returns
+    -------
+    is_img: boolean
+        True if get_data and get_affine methods are present and callable,
+        False otherwise.
+    """
+
+    # We use a try/except here because this is the way hasattr works
+    try:
+        get_data = getattr(obj, "get_data")
+        get_affine = getattr(obj, "get_affine")
+        return callable(get_data) and callable(get_affine)
+    except AttributeError:
+        return False
+
+
+def load_img(img):
+    """Load a niimg and check if it has required methods
+    """
+    if isinstance(img, basestring):
+        # data is a filename, we load it
+        img = nibabel.load(img)
+    elif not is_img(img):
+        raise TypeError("Data given cannot be converted to a nifti"
+                        " image: this object -'%s'- does not expose"
+                        " get_data or get_affine methods"
+                        % short_repr(img))
+    return img
+
+
+def new_img(data, affine, header=None):
```

Comment:
Once the header-related `new_img_like` (or however it will be called) becomes reality, the function name `new_img` will probably be looked for by users that know the former. Yes, this is `__future__` talk :) but in that light this addition seems very much consistent.